test/object-store/sync/app.cpp

```diff
@@ -971,17 +971,14 @@ TEST_CASE("app: call function", "[sync][app]") {
                                      CHECK(!error);
                                  });
 
-    app->call_function<int64_t>("sumFunc", {1, 2, 3, 4, 5},
-                                [&](Optional<app::AppError> error, Optional<int64_t> sum) {
-                                    REQUIRE(!error);
-                                    CHECK(*sum == 15);
-                                });
-
-    app->call_function<int64_t>(tsm.app()->sync_manager()->get_current_user(), "sumFunc", {1, 2, 3, 4, 5},
-                                [&](Optional<app::AppError> error, Optional<int64_t> sum) {
-                                    REQUIRE(!error);
-                                    CHECK(*sum == 15);
-                                });
+    bson::BsonArray toSum(5);
+    std::iota(toSum.begin(), toSum.end(), static_cast<int64_t>(1));
```

Comment:
This makes sure that the array gets sent to the server as an array of `$numberLongs`. Just declaring a `BsonArray = { 1, 2, 3, 4, 5}` generates an array of `int32_t` BSONs, because the type of a literal `1` is an `int` which chooses the `int32_t` Bson constructor.